core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -568,27 +631,26 @@ class RemoteIndexCacheTest {
   }
 
   private def verifyFetchIndexInvocation(count: Int,
-                                         indexTypes: Seq[IndexType] =
-                                         Seq(IndexType.OFFSET, IndexType.TIMESTAMP, IndexType.TRANSACTION)): Unit = {
+                                         indexTypes: Seq[IndexType]): Unit = {
```

Comment:
I don't think this change is necessary. When fetchIndex, we'll always fetch all 3 types of indexes. So why can't we verify all of them by default?

Reply:
Addressed this comment here: https://github.com/apache/kafka/pull/14483#discussion_r1347046103